core/cmd/eth_keys_commands.go

```diff
@@ -100,10 +100,6 @@ func initEthKeysSubCmd(s *Shell) cli.Command {
 						Usage:    "chain ID of the key",
 						Required: true,
 					},
-					cli.Uint64Flag{
```

Comment:
So we are disabling this, because setting nonce should only be done by the nonce-syncer after verifying from onchain?

Reply:
Right that's my understanding. The next nonce is only determined by either transactions in our `evm.txes` table or what's on-chain. If there's any sync issues, the nonce-syncer would automatically set the correct nonce so there shouldn't be a need to manually set. But if I'm missing a scenario and we need to allow a manual updates, I can try to add it back although it'd require eth_keys_controller to update the in-memory map in the broadcaster

Reply:
IIRC this was primarily used to 'fix' the nonce when it was set too high (perhaps after deleting some local txes which never went out to the pool) - in which case we shouldn't need it any more. However, I wonder if there were other edge cases :thinking:
cc @samsondav

Reply:
Yes, logically we should NEVER need someone to manually set a nonce.
The TXM's pending Txs, and onchain nonce value can be used exclusively to determine what the next nonce to use should be.
But just want to be wary of any legacy things we break here.